File: ToyLanguageExam2/src/Model/VarExp.java
package Model;

public class VarExp extends Exp {
    private String id;

    public VarExp(String id) {
        this.id = id;
    }

    int eval(MyIDictionary<String,Integer> tbl,MyIHeap<Integer,Integer> hp) throws MyException {
        try {
            return tbl.lookup(id);
        }
        catch (MyException ex)
        {
            throw new MyException(""+ex);
        }
    }

    @Override
    public String toString() {
        return id;
    }
}

File: ToyLanguage7/ToyLanguage7/VarExp.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ToyLanguage7
{
    class VarExp : Exp
    {
        private String id;

        public VarExp(String id)
        {
            this.id = id;
        }


        public override int eval(MyIDictionary<string, int> tbl)
        {
            try
            {
                return tbl.lookup(id);
            }
            catch (MyException ex)
            {
                throw new MyException("" + ex);
            }
        }

        public override string ToString()
        {
            return id;
        }
    }
}
